src/ProjectTempoListener.cpp

```diff
@@ -27,12 +31,14 @@ static const AttachedProjectObjects::RegisteredFactory key {
 
 ProjectTempoListener::ProjectTempoListener(
    AudacityProject& project, TrackList& trackList)
-    : mTrackList { trackList }
+    : mProject { project }
+    , mTempo { ProjectTimeSignature::Get(project).GetTempo() }
```

Comment:
Minor: To be consistent with the way `mTrackList` is obtained, I'd have opted for adding a `mViewInfo` member variable and called `ViewInfo::Get` from the outside.

Reply:
Good idea